### PR TITLE
Rewrite to remove jQuery, do some cleanup, and add some new features

### PR DESCRIPTION
**Pull request by EllipticEllipsis**
This started off as implementing collapsable file boxes using only CSS instead of js or jQuery, but ended up with several other adjustments and improvements:

- Cleaned up the box borders, fixing issues like breaks caused by rounding not being removed, missing borders when the file box is closed, and the rule being visible when the file box is closed, adding a top border to the chart box
- Added `Total`s to the files when closed, for quick reference. I'm not completely happy with how these came out, but it's a start. Can change the wording or the number used if people have a particular preference for what to show here for their game.
- Switched to `flex` for the file box headers since it seems to be the easiest way of implementing the previous point, and seems a bit more natural.
- Added a glowy effect to file buttons on the progress page on hover. (But not to the index since they aren't meant to do anything on that page)
- Remove the lower-right rectangle next to the Matching button with clipping (browser support for this is apparently a bit limited: it's fine in Firefox, at least). This could also be improved by using more specific lengths for various elements: I suspect it's a bit fragile at present.
- Replace most of the `<br>`s with more robust spacing commands.
- **EDIT:** And change the font on the Main Page button to Chiaro like basically everything else.

The fileboxes now look like this on page load:
image

Collapsing is currently implemented using checkboxes so that several file boxes can be open at once; an effect closer to the current version could be achieved using radio buttons instead, but my own preference is the former.

There ~~are two things~~ is one thing that I couldn't get to work satisfactorily:
- ~~The charts are very stubborn about reflowing: I found none of the methods that were recommended to get them to adjust to the correct size actually did anything, so I resorted to the vile piece of javascript in the header, which closes all but the first file box after waiting long enough for them to load sufficiently that the charts collect the correct dimensions (but not long enough for this to be visible in a browser). If anyone can come up with a more sensible way to make them size correctly, I'm all ears.~~ Fixed it, it was due to not keeping side padding when in the closed state.
- The file boxes open on click with the div opening to full height immediately, followed by a fade-in effect intended to mimic the file select screen somewhat. I could get this open animation to work easily enough, but couldn't find a reasonable way to get the corresponding reverse animation for closing.

### CODE REVIEW

File: progress.html [hunk 1]
```diff
@@ -14,175 +14,213 @@
...
+        window.addEventListener('DOMContentLoaded', async (event) => {
+
+            await new Promise(r => setTimeout(r, 100));
+
+            var boxes = document.getElementsByTagName('input');
+            for (var i = 1; i < boxes.length; i++) {
+                boxes[i].checked = false;
+            }
+        });
+    </script>
```
Comment (AngheloAlf) [inline progress.html]: Shouldn't be better to have this script in `progress.js`?

File: progress.css [hunk 2]
```diff
@@ -3,4 +3,110 @@ div#progressChart {
...
+}
+
+input + .zelda-file-box label ~ * {
+  /* display: none; */
+  /* padding: 0 !important;
+  border-style: none !important; */
+  /* height: 0;
+  overflow: hidden;
+  opacity: 0; */
+  /* transition: max-height 0.4s ease; */
```
Comment (AngheloAlf) [inline progress.css]: I guess this should be removed.

File: progress.css [hunk 3]
```diff
@@ -3,4 +3,110 @@ div#progressChart {
...
+
+input:not(:checked) + .zelda-file-box label ~ * {
+  border-top-style: none;
+  border-bottom-style: none;
+  padding: 0;
+  height: 0;
+  overflow: hidden;
+  opacity: 0;
+  /* transition: max-height 0.4s ease; */
+  /* transition: opacity 1s ease; */
```
Comment (AngheloAlf) [inline progress.css]: And this

**Review by AngheloAlf — COMMENTED** [review]
This looks good, and removing jQuery is always a win.
Just a few minor things

File: progress.html (hunk 1 above)
Comment (EllipticEllipsis) [reply progress.html]: Looks like it works. I'll hide it in there...

**Comment by EllipticEllipsis** [comment]
Fixed the graph reflow issues: side padding is now preserved in the "closed" states, so the graphs always see the correct box width. This also corrects an issue that the current implementation has with reflow of the graphs in the closed state when resizing the window.

**Comment by ethteck** [comment]
LGTM! Any objections before I merge? I'll wait a few hours for people to wake up in the US

**Review by AngheloAlf — APPROVED** [review]

**Pull request merged**